### ISSUE TITLE
Parser mixing up characters.

### ISSUE THREAD

**Issue by daverickdunn** [steelbrain/linter#1547]
Atom 1.20.1 x64 Windows
Linter 2.2.0
Linter-ui-default 1.6.10

Hi, been getting these errors several times a day for the past few weeks (haven't been using this much longer than that so not sure if it's been since a particular update...). 

Not even sure if it's linter or linter ui...

Once the bug has occurred, further manipulation of the characters in the editor will cause even stranger output in the error message. i.e. here `forFeatureFeclarations` may become `arationsFeatfor` simply by backspacing a few characters.

![image](https://user-images.githubusercontent.com/7645242/30915717-880d648a-a38f-11e7-9a05-8c64d831627c.png)

Restarting Atom always resolves it. 

Seems to happen anywhere \ anytime, haven't noticed any sequence of events causing it, but will keep an eye out.

Cheers!

**Comment by Arcanemagus** [comment]
That doesn't have anything to do with `linter` or `linter-ui-default`, they have no code to manipulate text. Why do you feel it was one of those causing it?

**Comment by daverickdunn** [comment]
Because it's the Linter package that is reporting the error. 

Perhaps I didn't explain the issue very well:

1. I type: `forFeature()`. (This code is correct).

2. Linter states that this is an error, because clearly `forFeatureFeclarations` does not exist.

**`forFeature != forFeatureFeclarations`** 

Now I've little to no experience with the Atom API, quite possibly this is due to an Atom error that's occurring before the value ever reaches _Linter_, but from where I'm standing, with no in-depth knowledge of either code base, it appears to be a _Linter_ issue. Further, I personally have never seen this happen outside of Linter / Typescript and I've been using Atom everyday for the past 2 years or so.

**Comment by Arcanemagus** [comment]
Ah ha! The text in the editor itself isn't being changed, but only the message coming from the provider?

**Comment by daverickdunn** [comment]
Yes.

**Comment by Arcanemagus** [comment]
Can you run `Linter: Debug` from the Command Palette and report the list of matching providers here? This is a bug in one of those providers.

**Comment by daverickdunn** [comment]
Will do.

**Comment by daverickdunn** [comment]
Same issue, slightly different code point.

I launched Atom, and my first action was to rename a const at the top of a module from _red_ to _reduce_.
![image](https://user-images.githubusercontent.com/7645242/30955226-cbdba7d0-a42a-11e7-8fc3-f504fc245eca.png)

Then I did the same thing on the function parameter shown in the image below.
![image](https://user-images.githubusercontent.com/7645242/30955053-421203f0-a42a-11e7-9c57-de5c4e317c11.png)

Image is a bit small, but it subsequently reports an error : **Cannot find name 'reduceuce'**.
![image](https://user-images.githubusercontent.com/7645242/30955069-50bfe4b2-a42a-11e7-9202-9057e4822e10.png)

Both times I simply added _uce_ to the end of the var name. I didn't perform a single other operation within Atom. There were no code errors reported anywhere when I launched Atom.

Output from `Linter: Debug`:

```
Platform: win32
Atom Version: 1.20.1
Linter Version: 2.2.0
Opened file is ignored: No
Matching Linter Providers: 
Disabled Linter Providers: 
Standard Linter Providers: 
Indie Linter Providers: 
  - Typescript
UI Providers: 
  - Linter
Ignore Glob: **/*.min.{js,css}
VCS Ignored Paths are excluded: true
Current File Scopes: 
  - *
  - source.ts
  - meta.decorator.ts
  - meta.objectliteral.ts
  - meta.object.member.ts
  - meta.array.literal.ts
  - meta.brace.round.ts
```

**Comment by daverickdunn** [comment]
Just to add, this is more or less unusable now. Any code change causes Linter to report numerous errors, when in fact there are none. Again, relaunching Atom will remove the errors, until the next code change...

Also, I've tried disabling all but the following community packages:
```
atom-typescript: 11.0.9
busy-signal: 1.4.3
intentions: 1.1.5
linter: 2.2.0
linter-ui-default: 1.6.10
```

Gonna have to switch to VS Code for now unfortunately.

**Comment by Arcanemagus** [comment]
So this is _yet another_ bug in `atom-typescript`, I've seen nothing but problems with it myself. On the off chance you were interested in getting that fixed you should file an issue [over there](https://github.com/TypeStrong/atom-typescript).

> Switched to Atom IDE + ide-typescript.

Tying `ide-typescript` was going to be my next suggestion 😉, glad you found something that works!

**Comment by daverickdunn** [comment]
Cheers, not gonna bother bringing it up if they're experiencing a lot of bugs already, I'm sure it's a known issue. I must say though, it's been a blessing in disguise, Atom IDE with ide-typescript is great! Thanks again.
